doc/modules/grid_search.rst

```diff
@@ -167,6 +171,358 @@ variable that is log-uniformly distributed between ``1e0`` and ``1e3``::
       Random search for hyper-parameter optimization,
       The Journal of Machine Learning Research (2012)
 
+.. _successive_halving_user_guide:
+
+Searching for optimal parameters with successive halving
+========================================================
+
+Scikit-learn also provides the :class:`HalvingGridSearchCV` and
+:class:`HalvingRandomSearchCV` estimators that can be used to
+search a parameter space using successive halving [1]_ [2]_. Successive
+halving (SH) is like a tournament among candidate parameter combinations.
+SH is an iterative selection process where all candidates (the
+parameter combinations) are evaluated with a small amount of resources at
+the first iteration. Only some of these candidates are selected for the next
+iteration, which will be allocated more resources. What defines a resource is
+typically the number of samples to train on, but it can also be an arbitrary
+numeric parameter such as `n_estimators` in a random forest.
+
+As illustrated in the figure below, only a small subset of candidates
+'survive' until the last iteration. These are the candidates that have
+consistently ranked among the best candidates across all iterations. Each
```

Comment:
"best" is not true, I think, though I might be nitpicking here. It could have *just* made the cutoff in each round.  Maybe "was in the subset of winners" or was in the top half, though we haven't explained the halving yet here... hm

Reply:
I'll replace "best" by "top scoring":

> These are the candidates that have
consistently ranked among the top-scoring candidates across all iterations

LMK if that's not OK

Reply:
sounds good.